config/types/etcd.go

```diff
@@ -32,7 +33,9 @@ var (
 )
 
 // Options can be the options for any Etcd version
-type Options interface{}
+type Options interface {
```

Comment:
Why is this being added?

Reply:
This is hitting a weird edge case with the validation code. The validation code "flattens" all embedded structs, so its checking if you can call Validate() on the top level `Etcd` struct. Go's struct embeddeding rules allows that if an embedded member allows that, but only superficially (doesn't work with []T or map[T1]T2). Unfortunately (for us), go uses the interface supplied not the underlying data's types. 

Why not use the Validate<NAME>() style functions? The flattening *should* handle that just fine.